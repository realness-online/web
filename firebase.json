{
  "storage": {
    "rules": "storage.rules"
  },
  "hosting": {
    "public": "dist",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ],
    "headers": [
      {
        "source": "**",
        "headers": [
          {
            "key": "Content-Security-Policy",
            "value": "script-src 'unsafe-eval' 'unsafe-inline' 'self' https://www.google.com https://apis.google.com https://www.gstatic.com https://storage.googleapis.com/"
          }
        ]
      },
      {
        "source": "**",
        "headers": [
          {
            "key": "Cache-Control",
            "value": "max-age=31536000"
          }
        ]
      },
      {
        "source": "/service.worker.js",
        "headers": [
          {
            "key": "Cache-Control",
            "value": "no-cache"
          }
        ]
      }
    ]
  }
}
