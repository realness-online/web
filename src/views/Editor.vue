<script setup>
  import icon from '@/components/icon'
  import asFill from '@/components/posters/as-figure-fill'
  import asAnimation from '@/components/posters/as-animation'
  import asGrid from '@/components/posters/as-grid'

  import { Poster } from '@/persistance/Storage'
  import { useFullscreen, useMagicKeys } from '@vueuse/core'
  import { useRoute, useRouter } from 'vue-router'
  import { watch, computed, ref } from 'vue'

  const fill = ref(true)
  const stroke = ref(false)
  const animation = ref(false)
  const grid = ref(false)
  const route = useRoute()
  const router = useRouter()
  const itemid = `${localStorage.me}/${route.params.type}/${route.params.id}`

  const page_title = computed(() => {
    if (stroke.value) return 'Stroke'
    if (fill.value) return 'Fill'
    if (animation.value) return 'Animation'
    return 'Grid'
  })
  const color = computed(() => {
    if (stroke.value || fill.value) return true
    else return false
  })
  const back = () => {
    console.log('back')
    const me = localStorage.me.substring(2)
    const id = route.params.id
    const type = route.params.type
    router.replace({ path: '/posters', hash: `#${me}-${type}-${id}` })
  }
  const save = async () => {
    console.log('save')
    const poster = new Poster(itemid)
    await poster.save()
    back()
  }
  const { toggle: fullscreen, isFullscreen: is_fullscreen } = useFullscreen()
  const { f, enter, escape } = useMagicKeys()
  watch(enter, v => {
    if (v) save()
  })
  watch(escape, v => {
    if (v) back()
  })
  watch(f, v => {
    if (v) fullscreen()
  })
</script>
<template>
  <section id="editor" class="page">
    <header v-if="!is_fullscreen">
      <h1>{{ page_title }}</h1>
      <a class="fullscreen" @click="fullscreen"><icon name="fullscreen" /></a>
      <icon name="nothing" />
      <a @click="back"><icon name="remove" /></a>
      <a @click="save"><icon name="finished" /></a>
    </header>
    <as-fill v-if="fill" :itemid="itemid" />
    <as-animation v-if="animation" :itemid="itemid" />
    <as-grid v-if="grid" :itemid="itemid" />
    <footer>
      <menu>
        <icon :class="{ selected: color }" name="edit-color" />
        <icon :class="{ selected: animation }" name="animation" />
        <icon :class="{ selected: grid }" name="grid" />
      </menu>
    </footer>
  </section>
</template>
<style lang="stylus">
  section#editor
    &:fullscreen
    &:full-screen
      & > header > a.fullscreen
        visibility: hidden
    & > header
      align-items: center
      z-index: 2
      position: fixed;
      top: inset(top, base-line)
      left: inset(left)
      right: inset(right)
      padding: 0
      @media (min-width: pad-begins)
        border-radius: 1rem
        background: black-transparent
        padding: 0 (base-line * 0.5)
      & > h1
        margin: 0
        color: red
        position: relative
        z-index: 2
        text-shadow: 1px 1px 1px black-background
    & > header > a  > svg
    & > footer > menu > svg
      cursor: pointer
      fill: green
      .selected
        fill:red
      &:hover
        fill: red
      &.color > svg.opacity
        fill: black-background
        &:hover
          fill:transparent
      &.remove
      &.fullscreen
      &.finished
        fill-opacity: inherit
    & > footer > menu
      background-color: black-transparent
      position: fixed
      z-index: 2
      bottom: base-line
      left: base-line
      right: base-line
      display: flex
      justify-content: space-between
      & > svg
        position: relative;
        z-index: 2
        &.selected
          stroke: red
          fill red
        &.grid
          border: 1px solid green
          border-radius: base-line * 0.15
          transition: border-color
          &:hover
            transition: border-color
            fill: green
            border-color: red
</style>
