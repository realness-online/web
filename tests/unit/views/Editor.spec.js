import { shallowMount } from '@vue/test-utils'
import { getRouter } from 'vue-router-mock'
import firebase from 'firebase/app'
import 'firebase/auth'
import Editor from '@/views/Editor'
const user = { phoneNumber: '16282281824' }
describe('@/views/Editor.vue', () => {
  describe('Renders', () => {
    it('Shows an editor for a poster or an avatar', async () => {
      firebase.user = user
      localStorage.me = `/+${user.phoneNumber}`
      getRouter().setParams({ id: '559666932867' })
      const wrapper = shallowMount(Editor)
      expect(wrapper.element).toMatchSnapshot()
      firebase.user = null
    })
  })
})
