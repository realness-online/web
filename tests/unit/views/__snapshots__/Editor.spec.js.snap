// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/views/Editor.vue Renders Shows an editor for a poster or an avatar 1`] = `
<section
  class="page"
  id="editor"
>
  <header>
    <h1>
      Fill
    </h1>
    <a
      class="fullscreen"
    >
      <icon-stub
        name="fullscreen"
      />
    </a>
    <icon-stub
      name="nothing"
    />
    <a>
      <icon-stub
        name="remove"
      />
    </a>
    <a>
      <icon-stub
        name="finished"
      />
    </a>
  </header>
  <as-fill-stub
    itemid="/+16282281824/posters/559666932867"
  />
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  <footer>
    <menu
      style="visibility: hidden;"
    >
      <icon-stub
        class="selected"
        name="edit-color"
      />
      <icon-stub
        class=""
        name="animation"
      />
      <icon-stub
        class=""
        name="grid"
      />
    </menu>
  </footer>
</section>
`;
